### PR TITLE
Update refs to the latest tag on Release

### PR DESCRIPTION
**Pull request by NikitaSkrynnik**
Signed-off-by: Nikita Skrynnik <nikita.skrynnik@xored.com>

## Issue
https://github.com/networkservicemesh/deployments-k8s/issues/4804

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -34,9 +34,35 @@ jobs:
...
+
+      - name: Update references
+        run: |
+          sed -r -i 's/(ref=[a-z0-9]*)/ref='"${{ steps.get-tag-step.outputs.tag }}"'/g' `grep 'ref=' -rl *`
+          sed -r -i '/raw.githubusercontent.com/ s/(deployments-k8s\/[a-z0-9]*)/deployments-k8s\/'"${{ steps.get-tag-step.outputs.tag }}"'/g' `grep 'raw.githubusercontent.com' -rl *`
+          git config --global user.email "nsmbot@networkservicmesh.io"
+          git config --global user.name "NSMBot"
+          git add -- .
+          git commit -s -m "Update references to the latest tag"
+          git push
```
Comment (denis-tingaikin) [inline .github/workflows/release.yaml:61]: Can we push to the concrete branch?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -34,9 +34,35 @@ jobs:
...
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          ref: ${{github.event.ref}}
+
+      - name: Get tag
+        run: |
+          branch=${{github.event.ref}}
+          echo '::set-output name=tag::'${branch#refs/heads/release/}
+        id: get-tag-step
```
Comment (denis-tingaikin) [inline .github/workflows/release.yaml:51]: Is it possible to reuse it from another job?

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -34,32 +36,52 @@ jobs:
...
-          echo '::set-output name=tag::'${branch#refs/heads/release/}
-        id: get-tag-step
-      - name: Push tag ${{ steps.get-tag-step.outputs.tag }}
+      - name: Push tag ${{ needs.update-refs.outputs.tag }}
         run: |
           git status
-          git tag ${{ steps.get-tag-step.outputs.tag }}
-          git push origin ${{ steps.get-tag-step.outputs.tag }} -f
-      - name: Create release ${{ steps.get-tag-step.outputs.tag }}
+          git tag ${{ needs.update-refs.outputs.tag }}
```
Comment (denis-tingaikin) [inline .github/workflows/release.yaml:76]: `needs` looks not correct.

```suggestion
          git tag ${{ jobs.update-refs.outputs.tag }}
```

Comment (denis-tingaikin) [reply .github/workflows/release.yaml:76]: Apply this for other places.

**Review by denis-tingaikin — CHANGES_REQUESTED** [review]

**Comment by denis-tingaikin** [comment]
@NikitaSkrynnik Could you rebase this?

**Comment by NikitaSkrynnik** [comment]
@denis-tingaikin, rebase done

**Review by glazychev-art — APPROVED** [review]

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**